pkg/client/ocm/client.go

```diff
@@ -41,7 +41,8 @@ type Client interface {
 	DeleteCluster(clusterID string) (int, error)
 	ClusterAuthorization(cb *amsv1.ClusterAuthorizationRequest) (*amsv1.ClusterAuthorizationResponse, error)
 	DeleteSubscription(id string) (int, error)
-	FindSubscriptions(query string) (*amsv1.SubscriptionsListResponse, error)
+	FindSubscriptions(query string) ([]*amsv1.Subscription, error)
```

Comment:
What is the criteria on returning an AMSXXXResponse object vs an AMSXXX object?

Reply:
I believe we should always return the result instead of the response.
Anyway, here I had to change it because `amsv1.SubscriptionsListResponse` is not mockable, and that was preventing me from creating unit tests.

Reply:
Is it worth changing the response type of the GetSubscriptionByID below to return the resource object instead in that case?

Reply:
I believe all methods should return the resource object instead of the response object. However, most of them are currently producing the response one.

Let's discuss this in a tech meeting to get a decision and fix all of them if needed.